components/Activitylist.js

```diff
@@ -51,6 +87,17 @@ export default function Activitylist({
               <StyledDeleteButton onClick={() => handleDelete(activity.id)}>
                 <StyledTrashIcon icon={faTrash} />
               </StyledDeleteButton>
+              <StyledFavoriteButton
+                onClick={() => handleAddFavoriteButton(activity.id)}
+              >
+                {activity.userData.some((user) => user.userID === userID) && (
+                  <StyledFavoriteIconSolid icon={faHeartSolid} />
+                )}
+                {!activity.userData.some((user) => user.userID === userID) && (
```

Comment:
you could just render the regular heart as the "else" case from your first check with the `?` and `:` operators